### PR TITLE
Overview, Troubleshooting, and Permalinks

### PR DESCRIPTION
**Pull request by codydaig**
In terms of sections of the documentation, Overview and Troubleshooting were the only two that I could tell didn't need any changes to make then applicable to 0.4.0. The remaining sections have some sort of change to make them applicable. 

The navbar has been modified to show the link to 0.4.0 when you hover over it. I also changed the doc folder/file structure slightly to allow for permalinks. /docs.html will go to the current (stable) version which is marked 0.3.x in the code.

Lastly, I added permalinks for:
/docs/0.4.x/ which will render the docs for 0.4.x
/docs/0.3.x/ which will render the docs for 0.3.x (which is the current view)

I did this to allow for multiple versions of the docs on the website as the project continues to develop. 

This pull request is ready to be merged, however, I am submitting it as a discussion point for the Yo Generator Section and merging this can be delayed. (Which I am bringing up in #443 )

### CODE REVIEW

**Comment by lirantal** [comment]
@codydaig is there a way I can visually see the changes that this PR adds?

**Comment by codydaig** [comment]
http://codydaig.github.io/mean/

DO NOT MERGE THIS PULL REQUEST UNTIL I ADD THE CNAME BACK AND MODIFY CONFIG.YML. 

gh-pages makes it difficult to test on live sites other than the one you want it published too. If you want to test locally, you have to install jekyll and run jekyll serve from the directory. (I put the install and run commands in the README.md file of the gh-pages branch). So, for these bigger changes, I'll just keep it live on my gh-pages branch until it's ready to be merged.

**Comment by codydaig** [comment]
@ilanbiala @lirantal Thoughts on this structure to differentiate 0.4.x from 0.3.x before I continue adding documentation?

**Comment by lirantal** [comment]
LGTM
@ilanbiala @rhutchison?

**Comment by ilanbiala** [comment]
I think it's fine.

File: _config.yml [hunk 1]
```diff
@@ -2,8 +2,10 @@
 title: MEAN.JS
 email: support@meanjs.org
 description: > # Full-Stack JavaScript Using MongoDB, Express, AngularJS, and Node.js
-baseurl: "" # the subpath of your site, e.g. /blog/
-url: "http://meanjs.org/" # the base hostname & protocol for your site
+baseurl: "" # For the Live MEAN WEBSITE
+url: "http://meanjs.org/" # FOR THE LIVE MEAN WEBSITE
+baseurl: "/mean" # FOR TESTING ONLY
+url: "http://codydaig.github.io/" # FOR TESTING ONLY
```
Comment (rhutchison) [inline _config.yml]: Should this be here?

Comment (ilanbiala) [inline _config.yml]: @codydaig did you forget to delete lines 7 and 8?

File: _includes/css/common.css [hunk 2]
```diff
@@ -64,4 +64,10 @@
 	left: 0;
 	width: 100%;
 	height: 100%;
+}
+
+@media (min-width: 979px) {
+  ul.nav li.dropdown:hover > ul.dropdown-menu {
+    display: block;
+  }
 }
```
Comment (ilanbiala) [inline _includes/css/common.css]: Can we get a newline here?

File: _includes/docs/0.4.x/gettingstarted.html [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+  <p>
+    After the install process is over, you'll be able to run your application using Grunt, just run grunt default task:
+    <pre>$ grunt</pre>
+    Your application should run on the 3000 port so in your browser just go to <a target="_blank" href="http://localhost:3000">http://localhost:3000</a>.
+  </p>
+  <p>
+    That's it! your application should be running by now, to proceed with your development check the other sections in this documentation.
+    <br>If you encounter any problem try the <a href="#troubleshooting">Troubleshooting</a> section.
+  </p>
+</section>
```
Comment (ilanbiala) [inline _includes/docs/0.4.x/gettingstarted.html]: Newline

File: _includes/docs/0.4.x/overview.html [hunk 4]
```diff
@@ -5,5 +5,31 @@
...
+    <dt>Node.js</dt>
+    <dd>
+      Start by going through <a target="_blank" href="http://nodejs.org">Node.js Official Website</a> and this <a target="_blank" href="http://stackoverflow.com/questions/2353818/how-do-i-get-started-with-node-js">StackOverflow Thread</a>, which should get you going with the Node.js platform in no time.
+    </dd>
+  </dl>
+  When you're done with those resources and feel you understand the basic principals continue to other sections.
+  <br>
+  <br>Enjoy &amp; keep us updated,
+  <br>The MEAN.JS Team.
 </section>
```
Comment (ilanbiala) [inline _includes/docs/0.4.x/overview.html]: newline

File: _includes/docs/0.4.x/troubleshooting.html [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+    </dd>
+    <dt>Error: failed to connect to [localhost:27017]</dt>
+    <dd>
+      If you use a local MongoDB check to see if its running. In case you use an external service verify the URI in the <a href="#configuration">configuration files</a>.
+    </dd>
+  </dl>
+  <p class="alert alert-warning">
+    Sometimes it can be a local issue or something that we didn't cover, in case you have any further questions please contact us via the community page.
+  </p>
+</section>
```
Comment (ilanbiala) [inline _includes/docs/0.4.x/troubleshooting.html]: newline

File: docs.html [hunk 6]
```diff
@@ -3,102 +3,4 @@
...
-
-    {% include docs/0.3.x/angularjsmodules.html %}
-
-    {% include docs/0.3.x/menus.html %}
-
-    {% include docs/0.3.x/articleexample.html %}
-
-  </section>
-</section>
+{% include docs/0.3.x/index.html %}
```
Comment (ilanbiala) [inline docs.html]: newline

**Comment by ilanbiala** [comment]
Also, should the _CNAME be changed back to CNAME?

File: _config.yml (hunk 1 above)
Comment (codydaig) [inline _config.yml]: That is for testing so you can view it on my github pages site.

**Comment by codydaig** [comment]
@ilanbiala - The _CNAME will be changed back before the pull request is ready. That is for testing so you can view it on my github pages site.

**Comment by lirantal** [comment]
if we fixed the newline stuff and nothing to add then I'm ok with merging.
@codydaig squash commits please when we're done here?

**Comment by codydaig** [comment]
Newline changes have been made and commits have been squashed.

File: _config.yml (hunk 1 above)
Comment (lirantal) [inline _config.yml]: what's the issue there? @codydaig

Comment (codydaig) [inline _config.yml]: @lirantal ??

**Comment by codydaig** [comment]
@ilanbiala I made the new line changes and squashed my commits. Anything else or can this be merged for other people to start contributing?

**Pull request merged**